File: challenges/hacker/153/EpicJhon.php
<?php

// http://www.hacker.org/challenge/chal.php?id=153

$str = '93752xxx746x27x1754xx90x93xxxxx238x44x75xx08750912738x8461x8759383xx328x4x4935903x6x5550360535004x0xx945958961296x267x8842xxx5x6xx61x4x48482x80xxx83316843x7x4x83x9521731xxx25x51xx457x6x5x9698222x771237745034x5133592x27xx8x87xx35221x36x0x50x23x7x63x998418xx';

$sum = 0;
$i = 0;
do {
    if ($str[$i] == 'x') {
        $str = substr($str, 0, $i) . substr($str, $i + 1);
        $i -= 2;
    } else {
        $sum += intval($str[$i]);
        $i++;
    }
} while($i <= strlen($str));

echo $sum;

File: challenges/hacker/160/EpicJhon.py
# http://www.hacker.org/challenge/chal.php?id=160

hexa = '8776459cf37d459fbb7b5ecfbb7f5fcfb23e478aaa3e4389f378439aa13e4e96a77b5fc1f358439df36a4486a03e4381b63e5580a66c0c8ebd6d5b8aa13e459cf34e4d9fa67f02cf90714288a17f589abf7f5886bc705fcfbc700c96bc6b5ecfb7775f8cbc68499daa3f'

val = ''
count = 0
for b1 in range(0, 256):
    for b2 in range(0, 256):
        for b3 in range(0, 256):
            for b4 in range(0, 256):
                for i in range(0, len(hexa), 2):
                    count = count + 1

                    if count == 1:
                        temp = b1
                    if count == 2:
                        temp = b2
                    if count == 3:
                        temp = b3
                    if count == 4:
                        temp = b4

                    c = int(hexa[i:i+2], 16)
                    ch = chr(c^temp)
                    val += ch

                    if count == 4:
                        count = 0

                count = 0
                original = val
                val = val.lower()

                if len(val) != len(hexa)/2:
                    val = ''
                    break

                if 'this' in val:
                    print(original)
                if 'challenge' in val:
                    print(original)

                val = ''
